APMrover2/control_modes.cpp

```diff
@@ -156,8 +159,8 @@ void Rover::read_aux_switch()
                 return;
             }
 
-            // record the waypoint if in manual or steering modes
```

Comment:
Yes, there are some other cases where we should have more specific methods on the mode object as well.  Like an "allow_arming()".

I think I'll change this check so that it simply checks that we're not in auto mode.